Templates/DefaultComponent/Template/Source/${Name}Component.h

```diff
@@ -0,0 +1,39 @@
+// {BEGIN_LICENSE}
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+// {END_LICENSE}
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <${SanitizedCppName}Interface.h>
+
+namespace ${SanitizedCppName}
+{
+    class ${SanitizedCppName}Component
+        : public AZ::Component
+        , public ${SanitizedCppName}RequestBus::Handler
+    {
+    public:
+        AZ_COMPONENT(${SanitizedCppName}Component, "{${Random_Uuid}}");
+
+        static void Reflect(AZ::ReflectContext* context);
+
+        static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
+        static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+        static void GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent);
+
```

Comment:
Would it be worth super over commenting the code here and elsewhere, to really indicate what each method is for?  

comments like "Over here, fill out the body with what services you need, example this.  Your components will sort and etc"

For all the empty functions or default ones we provide?

SHould we provide more than is necessary with empty bodies for all the common stuff?

Reply:
+1

Reply:
Agreed. These are some of the things people have the most trouble with and questions about. Since this is a template, guiding them through the process, better to be verbose and easier to delete what they don't need than find out what they do.

Reply:
Great idea. I've added comments for each method, mostly straight out of the AZ Component header.

Reply:
Also added a TODO to register this component in your module interface.